scripts/hail_batch/eqtl_hail_batch/generate_eqtl_spearman.py

```diff
@@ -66,7 +66,7 @@ def run_spearman_correlation_scatter(
     geneloc_df = geneloc_df[geneloc_df.geneid.isin(gene_ids)]
     geneloc_df = geneloc_df.assign(left=geneloc_df.start - 1000000)
     geneloc_df = geneloc_df.assign(right=geneloc_df.end + 1000000)
-    geneloc_df.to_csv(output_prefix + f'_gene_SNP_pairs_{idx}.tsv')
+    geneloc_df.to_csv(output_prefix + f'_gene_SNP_pairs.tsv')
```

Comment:
Both `gene_SNP_pairs.tsv` and `log_residuals.tsv` only need to be run once (it's the same exact file for each idx), but the script breaks if I don't add in `idx`. Is there a smarter way to run this without having so many duplicated files?

Reply:
What do you mean same file for each `idx`? If it doesn't change per index, we shouldn't need to compute them at all in that function. Maybe we can refactor it out into it's own method?